### PR TITLE
Boostrap the `ctest` framework for future unit tests written in C

### PR DESCRIPTION
**Pull request by emturner**
Lifted from: https://github.com/bvdberg/ctest 
Luckily, integration is easy as it is also Apache 2.0 licenced!

### CODE REVIEW

**Review by ajinkyaraj-23 — COMMENTED** [review]
Can you add one sample test on how to use this ctest in given context?
How ctest will link with parser c files? where the test_x.c should be added?

File: Makefile [hunk 1]
```diff
@@ -73,12 +73,14 @@ clean:
...
-		test/samples/operations/nano/samples.hex\
+		test/samples/operations/nano/samples.hex	\
 		tests/unit/parser/*.ml				\
 		tests/unit/parser/*.[ch]			\
 		tests/unit/parser/dune				\
 		tests/unit/parser/Makefile			\
-		tests/unit/Makefile
+		tests/unit/Makefile				\
+		tests/unit/ctest/Makefile			\
+		tests/unit/ctest/*.[ch]
```
Comment (spalmer25) [inline Makefile:83]: Would it make sens / be possible to have those in the `tests/unit/Makefile` ?

**Review by spalmer25 — COMMENTED** [review]
The `.gitignore` in `tests/unit/` could also be moved to `tests/unit/parser/`.

**Review by ajinkyaraj-23 — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (emturner) [reply Makefile]: I'm not completely sure why this was needed either - could we merge as is and then modify in future?

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**